cmd/tag-set.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * MinIO Client (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/minio/cli"
+	json "github.com/minio/mc/pkg/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio/pkg/console"
+)
+
+var tagSetCmd = cli.Command{
+	Name:   "set",
+	Usage:  "set tags for an object",
+	Action: mainSetTag,
+	Before: setGlobalsFromContext,
+	Flags:  globalFlags,
+	CustomHelpTemplate: `Name:
+	{{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [COMMAND FLAGS] TARGET [TAGS]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+DESCRIPTION:
+   Assign object tags (key,value) to target.
+
+EXAMPLES:
+  1. Assign the tags to an existing object.
+     {{.Prompt}} {{.HelpName}} s3/testbucket/testobject "key1=value1&key2=value2&key3=value3"
+
+`,
+}
+
+// tagSetTagMessage structure will show message depending on the type of console.
+type tagSetMessage struct {
+	Status string `json:"status"`
+	Name   string `json:"name"`
+	Error  error  `json:"error,omitempty"`
+}
+
+// tagSetMessage console colorized output.
+func (t tagSetMessage) String() string {
+	return console.Colorize(tagPrintMsgTheme, "Tags set for "+t.Name+".")
+}
+
+// JSON tagSetMessage.
+func (t tagSetMessage) JSON() string {
+	msgBytes, e := json.MarshalIndent(t, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+	return string(msgBytes)
+}
+
+func parseTagSetMessage(tags string, urlStr string, err error) tagSetMessage {
```

Comment:
```suggestion
func parseTagSetMessage(urlStr string, err error) tagSetMessage {
```

Reply:
The name should not be `parse...` you are not parsing anything, you are setting up values in a structure